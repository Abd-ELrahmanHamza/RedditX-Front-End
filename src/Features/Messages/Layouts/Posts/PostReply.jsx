import { useEffect, useState } from "react";
import PostReplayItem from "../../Components/PostReply/PostReplyItem";
import { PageContainer, EmbeddedPage, Empty, EmptyMessage } from "./PostReply.styled";

const messagesData = [
  {
    aurthor: "Mohamed",
    title: "Greeting",
    time: "2022-11-29T",
    msg: "Hello Hello",
    upvote: "neutral",
    admin: true,
    read: false,
    id: 1,
    block: false,
  },
  {
    aurthor: "Ahmed",
    title: "Mod",
    time: "2022-11-29T",
    msg: "You are Mod",
    upvote: "neutral",
    admin: false,
    read: false,
    id: 2,
    block: false,
  },
];

/**
 * Component that contains the Post Replies Panel
 *
 * @Component
 * @returns {React.Component}
 */
function PostReplay({replies}) {
  const [eachMessage, setEachMessage] = useState(messagesData);
  let Message = (
    <Empty>
      <EmptyMessage>
        there doesn't seem to be anything here
      </EmptyMessage>
    </Empty>
  );
  
  useEffect(()=>{
    if(replies&& replies.messages &&replies.messages.length!==0) {
      setEachMessage(replies.messages);
    }
  }, [replies]);

  if(eachMessage && eachMessage.length!==0){
    Message = eachMessage.map((item) => {
      if(!item.isDeleted) {
      return (
        <PostReplayItem
          changeMessage={setEachMessage}
          aurthor={item.aurthor}
          title={item.title}
          time={item.time}
          msg={item.msg}
          upvote={item.upvote}
          admin={item.admin}
          read={item.read}
          id={item.id}
          block={item.block}
          key={item.id}
        />
      );}
    });
  }

  return (
    <EmbeddedPage>
      <PageContainer>{Message}</PageContainer>
    </EmbeddedPage>
  );
}

export default PostReplay;
