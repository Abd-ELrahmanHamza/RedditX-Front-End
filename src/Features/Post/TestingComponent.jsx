// Import contexts providers
import { SubmitDestinationProvider } from "Features/Post/Contexts/submitDestination";
import { CreatePostTitleProvider } from "Features/Post/Contexts/createPostTitle";
import { CreatePostFlagsProvider } from "Features/Post/Contexts/createPostFlags";
import { CreatePostAttachmentsProvider } from "Features/Post/Contexts/createPostAttachments";
import { CreatePostFlairsProvider } from "Features/Post/Contexts/createPostFlairs";
import { CreatePostTextProvider } from "Features/Post/Contexts/createPostText";
import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  const auth = useAuth();
  useEffect(() => {
    auth.login({
      expiresIn: 86400,
      token:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbFR5cGUiOiJiYXJlIGVtYWlsIiwidXNlcm5hbWUiOiJ0Ml9ob21lIiwiaWF0IjoxNjY5OTI0MDEzLCJleHAiOjE2NzAzNTYwMTN9.FDa59vJi5gWy4dQVkEtgyCfm3Pp2BtHkPz0_AfNVKgY",
      username: "t2_home",
    });
  }, []);

  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SubmitDestinationProvider>
          <CreatePostTitleProvider>
            <CreatePostFlagsProvider>
              <CreatePostAttachmentsProvider>
                <CreatePostFlairsProvider>
                  <CreatePostTextProvider>{children}</CreatePostTextProvider>
                </CreatePostFlairsProvider>
              </CreatePostAttachmentsProvider>
            </CreatePostFlagsProvider>
          </CreatePostTitleProvider>
        </SubmitDestinationProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
