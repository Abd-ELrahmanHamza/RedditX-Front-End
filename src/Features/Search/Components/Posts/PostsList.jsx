import PostItem from "./Post";
import { ContainerPostsList } from "./PostsList.styled";
/**
 * Component that contains the Postslist component and the PostslistItems.
 *
 * @Component
 * @returns {React.Component}
 */
const PostsList = () => {
  return (
    <ContainerPostsList>
      <div className="Posts-List">
        <div className="Sub-List">
          <PostItem />
          <PostItem />
          <PostItem />
          <PostItem />
          <PostItem />
        </div>
      </div>
    </ContainerPostsList>
  );
};

export default PostsList;
