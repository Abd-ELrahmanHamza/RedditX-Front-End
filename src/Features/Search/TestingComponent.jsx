// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";
// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { SafeContextProvider } from "./Contexts/SafeSearchContext/Safe-context";
import { SearchContextProvider } from "./Contexts/SearchWordContext/Search-context";

const TestingComponent = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SearchContextProvider>
          <SafeContextProvider>{children}</SafeContextProvider>
        </SearchContextProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default TestingComponent;
