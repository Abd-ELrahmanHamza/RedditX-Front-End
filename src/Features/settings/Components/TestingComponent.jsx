import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        {children}
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
