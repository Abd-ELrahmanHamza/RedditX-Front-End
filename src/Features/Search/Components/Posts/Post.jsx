import PostFooter from "../PostFooter/PostFooter";
import PostHeader from "../PostHeader/PostHeader";
import { ContainerPost } from "./Post.styled";

/**
 * Component that contains the PostslistItems included PostHeader Component and post body and PostFooter Component.
 *
 * @Component
 * @returns {React.Component}
 */

const PostItem = () => {
  return (
    <ContainerPost>
      <div className="post-outline">
        <PostHeader />
        <div className="post-body">
          <div className="post-body-content">
            <a href="https://www.w3schools.com/">
              <div>
                <p>
                  John Oliver Refers to NFL’s ‘Monday Night Football’ as
                  “Primetime Programming Where People Kill Themselves for
                  Entertainment”
                </p>
              </div>
            </a>
            <span className="flair">flair</span>
          </div>
          <div className="post-image">
            <div className="img">
              <a href="https://www.w3schools.com/">
                <div className="image"></div>
              </a>
            </div>
          </div>
        </div>
        <PostFooter />
      </div>
    </ContainerPost>
  );
};

export default PostItem;
