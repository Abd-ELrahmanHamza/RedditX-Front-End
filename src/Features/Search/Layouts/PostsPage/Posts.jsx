import { ContainerPosts, StyledDropdown } from "./Posts.styled";
import Dropdown from "react-bootstrap/Dropdown";
// import DropdownButton from "react-bootstrap/DropdownButton";
import PostsList from "../../Components/Posts/PostsList";
import TopCommunities from "../../Components/CommunityCard/TopCommunities";
import CreateCommunity from "../../Components/CreateCommunity/CreateCommunity";
// import { PeopleContainer } from "../../Components/People/People.styled";
import People from "../../Components/People/People";
const Posts = () => {
  return (
    <>
      <ContainerPosts>
        <div className="subnav">
          <div className="sort">
            <StyledDropdown
              id="dropdown-basic-button"
              title="Sort"
              variant="transparent"
            >
              <Dropdown.Item href="#/action-1">Relevance</Dropdown.Item>
              <Dropdown.Item href="#/action-2">Hot</Dropdown.Item>
              <Dropdown.Item href="#/action-3">Top</Dropdown.Item>
              <Dropdown.Item href="#/action-3">New</Dropdown.Item>
              <Dropdown.Item href="#/action-3">Most comments</Dropdown.Item>
            </StyledDropdown>
          </div>
          <div className="time">
            <StyledDropdown
              id="dropdown-basic-button"
              title="time"
              variant="transparent"
            >
              <Dropdown.Item href="#/action-1">All time</Dropdown.Item>
              <Dropdown.Item href="#/action-2">Past Year</Dropdown.Item>
              <Dropdown.Item href="#/action-3">Past Mounth</Dropdown.Item>
              <Dropdown.Item href="#/action-3">Past week</Dropdown.Item>
              <Dropdown.Item href="#/action-3">Most 24 Hours</Dropdown.Item>
              <Dropdown.Item href="#/action-3">Most Hour</Dropdown.Item>
            </StyledDropdown>
          </div>
        </div>
        <div className="x">
          <PostsList />
          <div className="side-cards">
            <TopCommunities />
            <People />
            <CreateCommunity />
          </div>
        </div>
      </ContainerPosts>
    </>
  );
};

export default Posts;
