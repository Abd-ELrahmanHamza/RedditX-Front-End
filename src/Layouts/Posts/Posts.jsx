// import TopCommunities from "Features/explore/Components/CommunityCard/TopCommunities";
// import TopCommunities from "Features/explore/Components/CommunityCard/TopCommunities";
// import TopCommunities from "Components/CommunityCard/TopCommunities";
import CategoryTitle from "Components/CategoryTitle/CategoryTitle";
import TopCommunities from "Components/CommunityCard/TopCommunities";
import PostsList from "Components/PostsList/PostsList";
// import PostsList from "Features/explore/Components/PostsList/PostsList";
// import TopCommunities from "Features/Search/Components/CommunityCard/TopCommunities";
import { Cards, Layout, PostsPart } from "./Posts.styled";

const Posts = ({ CommunityList, communityListSub, PostList }) => {
  if (CommunityList) {
    return (
      <>
        <PostsPart>
          <CategoryTitle isPosts={true} />
          <PostsList PostList={PostList} />
        </PostsPart>
        <Cards>
          <Layout>
            <TopCommunities
              CommunityList={CommunityList}
              communityListSub={communityListSub}
            />
          </Layout>
        </Cards>
      </>
    );
  }
};

export default Posts;
