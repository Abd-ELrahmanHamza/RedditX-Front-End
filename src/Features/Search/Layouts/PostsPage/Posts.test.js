import Posts from "./Posts";
import { render, screen } from "@testing-library/react";
import TestingComponent from "Features/Post/TestingComponent";
it("this is a test for Posts", () => {
  render(
    <TestingComponent>
      <Posts />
    </TestingComponent>
  );
});
