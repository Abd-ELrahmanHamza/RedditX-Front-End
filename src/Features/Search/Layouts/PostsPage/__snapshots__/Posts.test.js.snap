// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Page this is a test for Posts Page 1`] = `ShallowWrapper {}`;

exports[`Search Page this is a test for Search Page 1`] = `ShallowWrapper {}`;
